tests/Unit/TaskList/CompleteSetupTaskTest.php

```diff
@@ -0,0 +1,99 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\Tests\Unit\tasks;
+
+use Automattic\WooCommerce\Admin\Features\OnboardingTasks\TaskLists;
+use Automattic\WooCommerce\GoogleListingsAndAds\MerchantCenter\MerchantCenterService;
+use Automattic\WooCommerce\GoogleListingsAndAds\TaskList\CompleteSetupTask;
+use Automattic\WooCommerce\GoogleListingsAndAds\Tests\Framework\UnitTest;
+use PHPUnit\Framework\MockObject\MockObject;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class CompleteSetupTaskTest
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\Tests\Unit\TaskList
+ */
+class CompleteSetupTaskTest extends UnitTest {
+
+	/** @var MerchantCenterService $merchant_center */
+	protected $merchant_center;
+
+	/** @var CompleteSetupTask $task */
+	protected $task;
+
+	/**
+	 * Runs before each test is executed.
+	 */
+	public function setUp(): void {
+		parent::setUp();
+		// Mock tasks list.
+		TaskLists::clear_lists();
+		TaskLists::add_list( [ 'id' => 'extended' ] );
+		// Mock MC setup.
+		$this->merchant_center = $this->createMock( MerchantCenterService::class );
+
+		$this->task = new CompleteSetupTask();
+		$this->task->set_merchant_center_object( $this->merchant_center );
+	}
+
+	/**
+	 * Test that after the task is registered and `init` fired,
+	 * the task is added to the `extended` list and the `add_woocommerce_extended_task_list_item` action is called.
+	 */
+	public function test_register() {
+		$this->task->register();
+		do_action( 'init', 1 );
+
+		$this->assertGreaterThan( 0, did_action( 'add_woocommerce_extended_task_list_item' ) );
+
+		$this->assertNotNull( TaskLists::get_list( 'extended' )->get_task( $this->task->get_id() ) );
+		$this->assertEquals( $this->task, TaskLists::get_list( 'extended' )->get_task( $this->task->get_id() ) );
+	}
```

Comment:
As mentioned in the comment above we shouldn't register and call init a second time here. Instead we should just test that it was registered and the hook was called.
The task is never going to be an identical object so at most I'd compare the ID or just leave it at assertNotNull since we are already fetching a specific ID so if that returns something it should always be the right one.

```suggestion
	public function test_register() {
		$this->assertEquals( 1, did_action( 'add_woocommerce_extended_task_list_item' ) );
		$this->assertNotNull( TaskLists::get_list( 'extended' )->get_task( $this->task->get_id() ) );
		$this->assertEquals(
			$this->task->get_id(),
			TaskLists::get_list( 'extended' )->get_task( $this->task->get_id() )->get_id()
		);
	}
```

Reply:
Maybe we should not create any instance but test the behavior of the already registered one 
https://github.com/woocommerce/google-listings-and-ads/pull/2026/commits/1c7b5fba988d0265e99b7dae5122b768e6462153